app/controllers/InputScreenController.scala

```diff
@@ -74,8 +74,14 @@ class InputScreenController @Inject() (
             )
 
             // setting defaults for missing fields
+            val isLeapYear = form.accountingPeriodStartDate.getYear % 4 == 0
```

Comment:
I dont see the need for this logic. form.accountingPeriodStartDate.plusYears(1).minusDays(1) should take care of this, we dont need to treat leap year any special

```scala> val l1 = LocalDate.of(2023,6,1)
l1: java.time.LocalDate = 2023-06-01

scala> l1.plusYears(1).minusDays(1)
res0: java.time.LocalDate = 2024-05-31```